### DISCUSSION TITLE
全局设置里接口问题

### DISCUSSION THREAD

**Discussion by xiaoyang86** [hiroi-sora/Umi-OCR#666] [General]
大佬问您一个小白的问题就是我在调试的时候有设置里面有好几个文字识别接口 但是解压缩后就剩一个rapidocr接口了，求助 谢谢大佬  万分感谢您

**Comment by xiaoyang86** [comment]
![Uploading 接口1.jpg…]()

**Comment by xiaoyang86** [comment]
![Uploading 打包错误信息.jpg…]()

**Comment by xiaoyang86** [comment]
删除temp
开始拷贝基础文件       
    Umi-OCR.exe        
    UmiOCR-data/i18n   
    UmiOCR-data/plugins
    UmiOCR-data/py_src
    UmiOCR-data/qt_res
    UmiOCR-data/runtime
    UmiOCR-data/site-packages
    UmiOCR-data/main.py
    UmiOCR-data/RUN_CLI.bat   
    UmiOCR-data/RUN_GUI.bat   
    UmiOCR-data/test_speed.bat
    UmiOCR-data/about.json    
结束拷贝基础文件
清理缓存：0
文件总数：1837，总大小：499MB
读取版本信息
解析版本信息： 2 1 4  0
    Paddle 插件组缺失 win7_x64_PaddleOCR-json
解析插件信息： {'Rapid': {'name': 'Umi-OCR_Rapid_v2.1.4', 'plugins': ['win7_x64_RapidOCR-json']}}
开始创建插件区分： Rapid
    重新创建 ./release/Umi-OCR_Rapid_v2.1.4
    删除 ./release/Umi-OCR_Rapid_v2.1.4/UmiOCR-data/plugins/win7_x64_PaddleOCR-json_v141
开始创建插件7z压缩包： Rapid

7-Zip (r) 23.01 (x86) : Igor Pavlov : Public domain : 2023-06-20

Scanning the drive:
155 folders, 1778 files, 212487938 bytes (203 MiB)

Creating archive: ./release/Umi-OCR_Rapid_v2.1.4.7z

Add new data to archive: 155 folders, 1778 files, 212487938 bytes (203 MiB)


Files read from disk: 1771
Archive size: 101564844 bytes (97 MiB)
Everything is Ok

开始创建插件sfx自解压文件：
 copy /b "D:\项目代码目录\Umi-OCR\dev-tools\7z\7z.sfx"+"D:\项目代码目录\Umi-OCR\release\Umi-OCR_Rapid_v2.1.4.7z" "D:\项目代码目录\Umi-OCR\release\Umi-OCR_Rapid_v2.1.4.7z.exe"
D:\项目代码目录\Umi-OCR\dev-tools\7z\7z.sfx
D:\项目代码目录\Umi-OCR\release\Umi-OCR_Rapid_v2.1.4.7z
已复制         1 个文件。
打包结束。

**Comment by hiroi-sora** [comment]
你好。项目自带的一键打包脚本，设定是一个发布包（如 `Umi-OCR_Paddle_v2.1.4.7z.exe`）内只包含一个OCR插件（如 `PaddleOCR-json` ）。

如果你想一个发布包内包含所有插件，也很简单，执行一次打包脚本后， release 目录下有个 temp 文件夹，它就是含所有插件的完整软件包。将它改名为 `Umi-OCR` 之类，就可以拿去用了。

如果你想做成自解压文件（`.7z.exe`），可以使用 7z 压缩软件，勾选 **创建自释放程序** 。

**Reply by xiaoyang86** [reply]
好的 谢谢大佬
